### PR TITLE
Implement support for BP_NODE_PROJECT_PATH

### PR DESCRIPTION
**Pull request by fg-j**
Part of paketo-buildpacks/nodejs#327

Signed-off-by: Sophie Wigmore <swigmore@vmware.com>
Co-authored-by: Sophie Wigmore <swigmore@vmware.com>

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

File: fakes/application_detector.go [hunk 1]
```diff
@@ -7,23 +7,25 @@ type ApplicationDetector struct {
 		sync.Mutex
```
Comment (arjun024) [inline fakes/application_detector.go:7]: Redundant file?

File: node_application_finder.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+type NodeApplicationFinder struct{}
+
+func NewNodeApplicationFinder() NodeApplicationFinder {
+	return NodeApplicationFinder{}
+}
+
+func (n NodeApplicationFinder) Find(workingDir string) (string, error) {
+
```
Comment (arjun024) [inline node_application_finder.go:16]: So I am trying to think again about returning the absolute path here.
Does the spec give a guarantee that (1) the absolute path of workingDir during build, and (2) the absolute path of workingDir during launch would be equal?
If not, couldn't they be mounted a different points? If that's the case, we should possibly rely on relative paths.
Could you investigate?

**Review by arjun024 — CHANGES_REQUESTED** [review]

File: node_application_finder.go (hunk 2 above)
Comment (fg-j) [reply node_application_finder.go:16]: I'm asking for some clarification in CNB slack: https://buildpacks.slack.com/archives/CSFE49VGR/p1615320551022100

Comment (fg-j) [reply node_application_finder.go:16]: Given the above, I agree it's probably a better practice to use a relative path.

**Review by arjun024 — CHANGES_REQUESTED** [review]
Left a couple of comments. Other than that looks good to me

**Review by arjun024 — APPROVED** [review]

**Pull request merged**